### ISSUE TITLE
create_prompt_layout lacks argument for specifying editing mode

### ISSUE THREAD

**Issue by ndronen** [jonathanslenders/python-prompt-toolkit#432]
SearchToolbarControl has a vi_mode argument, but the default of vi_mode=False is effectively immutable for callers of create_prompt_layout, because create_prompt_layout lacks an argument for specifying the editing mode.  Consequently, vi mode users are shown an emacs prompt ("I-search: " or "I-search backward:") when searching their history.

**Comment by jonathanslenders** [comment]
True. This was actually done consiously. I thought that this would be the most user friendly visualisation if we have a multiline input. If there is some indentation in the left margin, like line numbers, then the "(reverse-i-search)" visualisation doesn't work that well. (Which is by the way exactly the same, regardless of whether the user is in Vi or Emacs mode. However, only for incremental search.)

I can always expose that argument, but I guess I need to rethink the search implementation, and have both incremental and non-incremental search. (If that's the correct name.) In Vi-mode, you can start searching by pressing "Control-R", or by pressing "/" in navigation mode.

Changing the visualisation to "/" for Vi mode would be confusing without first implementing the search accordingly.

Thanks for opening the issue!
